### PR TITLE
Create a Github Actions based CI

### PR DESCRIPTION
**Pull request by amaslenn**
Build steps took from
http://bgate.mellanox.com/jenkins/job/OpenShmem-pr__published_from_mellanox_lab/.
Original Jenkins's job uses the internal HPCX daily env module. To have it I pull relevant GA package.

### CODE REVIEW

**Comment by amaslenn** [comment]
@mike-dubman please take a look

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: CI
+
+on: [push]
```
Comment (mike-dubman) [inline .github/workflows/main.yml]: why not on pull-request? no1 will push into master directly

Comment (amaslenn) [reply .github/workflows/main.yml]: `push` is more generic: it will automatically check all pushed branches and PRs (and also merge to master).

**Comment by amaslenn** [comment]
@mike-dubman I don't see the Actions tab for this project, probably it should be enabled in repo settings (I don't have permissions to check).

**Pull request merged**